src/shared/clients/iamClient.ts

```diff
@@ -52,28 +51,14 @@ export class DefaultIamClient {
 
     /**
      * Attempts to verify if a role has the provided permissions.
-     * @param roleArn IAM.SimulatePrinicipalPolicyRequest
-     * @returns True if the role has the provided permissions. Undefined when the role is missing or the 'simulatePrincipalPolicy' call was unsuccessful.
      */
-    public async hasRolePermissions(request: IAM.SimulatePrincipalPolicyRequest): Promise<boolean | undefined> {
-        if (request.PolicySourceArn === undefined) {
-            return undefined
-        }
-        try {
-            const permissionResponse = await this.simulatePrincipalPolicy(request)
-            if (!permissionResponse || !permissionResponse.EvaluationResults) {
-                return undefined
-            }
-            for (const evalResult of permissionResponse.EvaluationResults) {
-                if (evalResult.EvalDecision !== 'allowed') {
-                    return false
-                }
-            }
-            return true
-        } catch (error) {
-            getLogger().error('iam: Error during policy simulation. Skipping permissions check. Error: %O', error)
-            return undefined
+    public async getDeniedActions(request: IAM.SimulatePrincipalPolicyRequest): Promise<IAM.EvaluationResult[]> {
+        const permissionResponse = await this.simulatePrincipalPolicy(request)
+        if (!permissionResponse.EvaluationResults) {
+            throw new Error('No evaluation results found')
         }
+
+        return permissionResponse.EvaluationResults.filter(r => r.EvalDecision !== 'allowed')
```

Comment:
👍  always a good idea to return more info instead of just a boolean, if it is cheap to do so. Allows callers to make better decisions and also surface better info to the user.